### PR TITLE
version(v0.13.0): Update agent to support k8s v1.20+

### PR DESCRIPTION
**Pull request by aesca1er**
Update agent to support Kubernetes v1.20+
Update changelog

issue #CIRC-8497

* Tags: v0.13.0 k8s-v1.20+

### CODE REVIEW

**Comment by dhaifley** [comment]
Also, this repo seems to have several outdated dependencies. It might be a good idea, if you are making a new release, to get those dependabot PR's tested and incorporated, if possible.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,429 +1,784 @@
...
 
-* upd: log error message on cn mismatch in tls verify
-* upd: lint issues
-* upd: add pre-build lint back in
-* add: build_lint.sh script
-* upd: lint config
-* fix: ensure probe and resource metrics collected in sequential mode
-* build(deps): bump github.com/spf13/viper from 1.10.0 to 1.10.1
-* build(deps): bump github.com/rs/zerolog from 1.26.0 to 1.26.1
+1.13.0
```
Comment (graefen) [inline CHANGELOG.md:2]: Looks like 0.13.0 and 1.13.0 are being used interchangeably...we should probably stick with 0.13.0, right?

**Comment by aesca1er** [comment]
> Also, this repo seems to have several outdated dependencies. It might be a good idea, if you are making a new release, to get those dependabot PR's tested and incorporated, if possible.

I think those should be addressed in a patch release following this one, I tend to prefer changing as few things as possible at a time.
Moving forward, I would like to merge these much faster -- I had no ability to test if these changes would break the agent prior to setting up the cluster in beltsville.

**Comment by aesca1er** [comment]
> I understand, and approve the version update in the deployment file, but why did you change the whole CHANGELOG to just be git commit messages, deleting all of the actual changelog entries? I understand wanting to have the commit ids in there and automating this and so forth. But, this isn't actually how the changelogs are used in a lot of our projects. They are generally messages to the user about the changes, not a listing of individual commit messaged. We can get that commit info in other ways. And, for internal projects, Eric uses these to compose the overall release notes for customers. I would not want to see a change like this for repos I work on, like GoAPI.

Agreed on not removing the existing changelog notes, however I do think the changelogs should be generated from commit messages.

Conventional commits provide a method for supplying changelog notes in the commit (using my modification of the `[gitcomm](https://github.com/aescaler-a0d/gitcomm)` utility, this is the `[summary]` field).
Doing so keeps documentation as DRY as possible ([example](https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-multi-paragraph-body-and-multiple-footers)).

Moving forward, the changelogs I generate will be comprised of these summaries -- I'll have to modify the template to do so.
Additionally, I'll replace the entries I erroneously removed and revise the commit history.

File: CHANGELOG.md (hunk 1 above)
Comment (aesca1er) [reply CHANGELOG.md:2]: yep

**Comment by dhaifley** [comment]
Hi Antonio. That seems fine for this repo, I don't work on it so I don't have an opinion on how its changelog should be structured. But, I would just keep in mind, I know that you like the conventional commits methodology, but we don't all work the same way, and don't have a company policy about using that. On many of our repos, Eric does use the change log to compose overall release notices for customers / users. I'm not sure if this repo is part of that. When I write commit messages, they are intended for developers working on the repo, not necessarily customers. And, if I wanted to see all the commits / commit messages that were included in a release, I could just pull up a compare for that release in GitHub. I would not need to make a separate file in the repo just to contain that information. What I do put in the changelog are short bullet summaries of the various user facing changes that have gone into a release, intended for consumption by the users. These are sometimes a summary of many individual commit messages. To be clear, I am not opposed to using conventional commit messages. I think it is an interesting idea, and I do like them. But, I would be cautious about implementing anything that forces that to be used, unless it becomes company policy for everyone to work that way. And, most importantly, we want to make sure that the release notes process that Eric Sproul uses would not be affected by such a change.

**Pull request closed without merging**